tests/seed_selection/seed_selection_suite_test.go

```diff
@@ -0,0 +1,373 @@
+	info := retrieveDatacenterInfo()
+	name := ""
+	for _, node := range info.Nodes {
+		if node.Rack == rack && node.Seed {
+			name = node.Name
+			break
+		}
+	}
+
+	Expect(name).ToNot(Equal(""))
+	return name
+}
+
+
+func MinInt(a, b int) int {
+	if a < b {
+		return a
+	}
+	return b
+}
+
+func checkThereAreAtLeastThreeSeedsPerDc(info DatacenterInfo) {
+	seedCount := 0
+
+	for _, node := range info.Nodes {
+		if node.Seed {
+			seedCount += 1
+		}
+	}
+
+	expectedSeedCount := MinInt(info.Size, 3)
+	Expect(seedCount >= expectedSeedCount).To(BeTrue(), 
+		"Expected there to be at least %d seed nodes, but only found %d.", 
+		expectedSeedCount, seedCount)
+}
+
+func checkThereIsAtLeastOneSeedNodePerRack(info DatacenterInfo) {
+	rackToFoundSeed := map[string]bool{}
+	for _, node := range info.Nodes {
+		if node.Seed {
+			rackToFoundSeed[node.Rack] = true
+		}
+	}
+	
+	for _, rackName := range info.RackNames {
+		value, ok := rackToFoundSeed[rackName]
+		Expect(ok && value).To(BeTrue(), "Expected rack %s to have a seed node, but none found.", rackName)
+	}
+}
+
+func checkDesignatedSeedNodesAreStartedAndReady(info DatacenterInfo) {
+	for _, node := range info.Nodes {
+		if node.Seed {
+			Expect(node.Started).To(BeTrue(), "Expected %s to be labeled as started but was not.", node.Name)
+			Expect(node.Ready).To(BeTrue(), "Expected %s to be ready but was not.", node.Name)
+		}
+	}
+}
+
+func checkCassandraSeedListsAlignWithSeedLabels(info DatacenterInfo) {
+	expectedSeeds := []string{}
+	for _, node := range info.Nodes {
+		if node.Seed {
+			expectedSeeds = append(expectedSeeds, node.IP)
+		}
+	}
+	sort.Sort(sort.StringSlice(expectedSeeds))
+
+	re := regexp.MustCompile(`[0-9]+[.][0-9]+[.][0-9]+[.][0-9]+`)
+	for _, node := range info.Nodes {
+		if node.Ready && node.Started {
+			k := kubectl.ExecOnPod(node.Name, "--", "nodetool", "getseeds")
+			output := ns.OutputPanic(k)
+			seeds := re.FindAllString(output, -1)
+			if node.Seed {
+				seeds = append(seeds, node.IP)
+			}
+			sort.Sort(sort.StringSlice(seeds))
+
+			Expect(seeds).To(Equal(expectedSeeds), "Expected pod %s to have seeds %v but had %v", node.Name, expectedSeeds, seeds)
+		}
+	}
+}
+
+func checkSeedConstraints() {
+	info := retrieveDatacenterInfo()
+	// There should be 3 seed nodes for every datacenter
+	checkThereAreAtLeastThreeSeedsPerDc(info)
+
+	// There should be 1 seed node per rack
+	checkThereIsAtLeastOneSeedNodePerRack(info)
+
+	// Seed nodes should not be down
+	checkDesignatedSeedNodesAreStartedAndReady(info)
+
+	// Ensure seed lists actually align
+	//
+	// NOTE: The following check does not presently work due to 
+	// the lag time between when we update a seed label and when
+	// that change is reflected in DNS. Since we reload seed lists
+	// right after upating the label, some cassandra nodes will
+	// likely end up with slight out-of-date seed lists. KO-375
+	//
+	// checkCassandraSeedListsAlignWithSeedLabels(info)
+}
+
+func duplicate(value string, count int) string {
```

Comment:
I'll take `Things that should have been included in the base libraries` for 100

Reply:
Would probably be worthwhile to extract this out into the `ginkgo_util` package, since it could be used for any test with more than 1 node